### ISSUE TITLE
ERPNext Desk stuck at Logo in Production

### ISSUE THREAD

**Issue by Fauzdar1** [frappe/erpnext#27920]
I've deployed ERPNext on two Ubuntu 20.04 servers using the[ same installation procedure](https://www.rosehosting.com/blog/how-to-install-erpnext-on-ubuntu-20-04) for both and it works in one but doesn't work in another.

I  had created [a thread](https://discuss.erpnext.com/t/erpnext-desk-stuck-at-logo-in-production/81143) on ERPNext forum but only one person replied asking for `bench build` which does nothing.

## Description of the issue
After successful installation and setting up on production (even on dev), and accessing ERPNext through the set domain, it works for the first setup and login but after first login, from second login, it gets stuck on the Logo, see [here](https://erp.unlockvelocity.com/), route - `/app`. 

## Solution I've tried are:
1. `bench update`.
2. `bench migrate`
3. `bench build`

## Solution that works:
Reinstalling the whole server and ERPNext on it. 

## Context information (for bug reports)

**Output of `bench version`**
```
5.0.0-dev
```

## Steps to reproduce the issue

1. Install ERPNext.
2. Do the initial setup and Login.
3. Logout and Re-Login

### Observed result
Doesn't work.

### Expected result
Should work.

### Stacktrace / full error message

![image](https://user-images.githubusercontent.com/62703211/136932287-c9392133-88b1-412a-985d-5091dce6a375.png)

## Additional information

OS version - Ubuntu 20.04
Installation method - [How to Install ERPNext on Ubuntu 20.04](https://www.rosehosting.com/blog/how-to-install-erpnext-on-ubuntu-20-04)

Extra point to note - I open the website on Incog so this [old but same thread](https://github.com/frappe/erpnext/issues/8368) won't help.

**Comment by ankush** [comment]
Post output of `bench build`.

Also, click on the file link in the first error to see where syntax error is present. 
![Screenshot 2021-10-12 at 3 15 01 PM](https://user-images.githubusercontent.com/9079960/136932947-313e82e7-4f3a-4b1a-959a-78cfb324e3d4.png)


Also: 

> 5.0.0-dev

avoid using development versions.

**Comment by Fauzdar1** [comment]
Bench build Log
```
erpnext@vmi647968:/opt/bench/erpnext$ bench build
WARN: bench is installed in editable mode!

This is not the recommended mode of installation for production. Instead, install the package from PyPI with: `pip install frappe-bench`

Retrieving assets...

✔ Downloaded Frappe assets from http://assets.frappeframework.com/3312deb97d95297e4f518c6628a51b7dd21d38b8.tar.gz

Extracting assets...

✔ Restored frappe/dist/css/email.bundle.AEUPTTFL.css
✔ Restored frappe/dist/css/report.bundle.Y635HGFG.css
✔ Restored frappe/dist/css/login.bundle.ONSGCWSP.css
✔ Restored frappe/dist/css/chat.bundle.2CO7LVQI.css
✔ Restored frappe/dist/css/web_form.bundle.MB35FBK2.css
✔ Restored frappe/dist/css/print.bundle.2SWQKUCJ.css
✔ Restored frappe/dist/css/website.bundle.NVJLACDR.css
✔ Restored frappe/dist/css/desk.bundle.GYS2ELT4.css
✔ Restored frappe/dist/js/desk.bundle.6DXIO5G7.js.map
✔ Restored frappe/dist/js/frappe-web.bundle.NEJRF7DV.js.map
✔ Restored frappe/dist/js/libs.bundle.7O6DCBW2.js.map
✔ Restored frappe/dist/js/build_events.bundle.N2YVEOEQ.js.map
✔ Restored frappe/dist/js/chat.bundle.3ZSGENSS.js
✔ Restored frappe/dist/js/data_import_tools.bundle.XY5LI33T.js.map
✔ Restored frappe/dist/js/form.bundle.O7GU536T.js.map
✔ Restored frappe/dist/js/dialog.bundle.YEILFA5V.js.map
✔ Restored frappe/dist/js/user_profile_controller.bundle.IEUMSNOB.js
✔ Restored frappe/dist/js/bootstrap-4-web.bundle.ACJ3KVM3.js
✔ Restored frappe/dist/js/list.bundle.WX66KTJL.js.map
✔ Restored frappe/dist/js/report.bundle.DP3L6LZI.js.map
✔ Restored frappe/dist/js/data_import_tools.bundle.XY5LI33T.js
✔ Restored frappe/dist/js/build_events.bundle.N2YVEOEQ.js
✔ Restored frappe/dist/js/dialog.bundle.YEILFA5V.js
✔ Restored frappe/dist/js/controls.bundle.BW2PLJ4N.js
✔ Restored frappe/dist/js/chat.bundle.3ZSGENSS.js.map
✔ Restored frappe/dist/js/checkout.bundle.PNWSN6RI.js.map
✔ Restored frappe/dist/js/barcode_scanner.bundle.K4AINWAD.js
✔ Restored frappe/dist/js/bootstrap-4-web.bundle.ACJ3KVM3.js.map
✔ Restored frappe/dist/js/controls.bundle.BW2PLJ4N.js.map
✔ Restored frappe/dist/js/checkout.bundle.PNWSN6RI.js
✔ Restored frappe/dist/js/form.bundle.O7GU536T.js
✔ Restored frappe/dist/js/recorder.bundle.QWTNBPXO.js.map
✔ Restored frappe/dist/js/list.bundle.WX66KTJL.js
✔ Restored frappe/dist/js/user_profile_controller.bundle.IEUMSNOB.js.map
✔ Restored frappe/dist/js/web_form.bundle.QABF5X3O.js.map
✔ Restored frappe/dist/js/libs.bundle.7O6DCBW2.js
✔ Restored frappe/dist/js/desk.bundle.6DXIO5G7.js
✔ Restored frappe/dist/js/barcode_scanner.bundle.K4AINWAD.js.map
✔ Restored frappe/dist/js/recorder.bundle.QWTNBPXO.js
✔ Restored frappe/dist/js/report.bundle.DP3L6LZI.js
✔ Restored frappe/dist/js/frappe-web.bundle.NEJRF7DV.js
✔ Restored frappe/dist/js/web_form.bundle.QABF5X3O.js
✔ Restored frappe/dist/css-rtl/login.bundle.UDH6MLYU.css
✔ Restored frappe/dist/css-rtl/email.bundle.6P2S2Y53.css
✔ Restored frappe/dist/css-rtl/print.bundle.42KHEBS5.css
✔ Restored frappe/dist/css-rtl/website.bundle.JIMEK3CT.css
✔ Restored frappe/dist/css-rtl/report.bundle.KEK267QG.css
✔ Restored frappe/dist/css-rtl/chat.bundle.H7NNYOVV.css
✔ Restored frappe/dist/css-rtl/web_form.bundle.6IU5HJ52.css
✔ Restored frappe/dist/css-rtl/desk.bundle.YDWNOXHN.css

Building missing assets...

Linking /opt/bench/erpnext/apps/frappe/node_modules to ./assets/frappe/node_modules                                                                             Linking /opt/bench/erpnext/apps/erpnext/node_modules to ./assets/erpnext/node_modules                                                                           ✔ Application Assets Linked


yarn run v1.22.15
$ node esbuild --production --files frappe/rtl_chat.bundle.css,frappe/rtl_desk.bundle.css,frappe/rtl_email.bundle.css,frappe/rtl_login.bundle.css,frappe/rtl_print.bundle.css,frappe/rtl_report.bundle.css,frappe/rtl_web_form.bundle.css,frappe/rtl_website.bundle.css --run-build-command
clean: postcss.plugin was deprecated. Migration guide:
https://evilmartians.com/chronicles/postcss-8-plugin-migration
clean: postcss.plugin was deprecated. Migration guide:
https://evilmartians.com/chronicles/postcss-8-plugin-migration
File                                                        Size

 DONE  Total Build Time: 984.19ms

Done in 1.95s.
Linking /opt/bench/erpnext/apps/frappe/node_modules to ./assets/frappe/node_modules                                                                             Linking /opt/bench/erpnext/apps/erpnext/node_modules to ./assets/erpnext/node_modules                                                                           ✔ Application Assets Linked


yarn run v1.22.15
$ node esbuild --production --skip_frappe --run-build-command
clean: postcss.plugin was deprecated. Migration guide:
https://evilmartians.com/chronicles/postcss-8-plugin-migration
clean: postcss.plugin was deprecated. Migration guide:
https://evilmartians.com/chronicles/postcss-8-plugin-migration
clean: postcss.plugin was deprecated. Migration guide:
https://evilmartians.com/chronicles/postcss-8-plugin-migration
clean: postcss.plugin was deprecated. Migration guide:
https://evilmartians.com/chronicles/postcss-8-plugin-migration
clean: postcss.plugin was deprecated. Migration guide:
https://evilmartians.com/chronicles/postcss-8-plugin-migration
clean: postcss.plugin was deprecated. Migration guide:
https://evilmartians.com/chronicles/postcss-8-plugin-migration
File                                                        Size

erpnext/dist/js/
├─ bank-reconciliation-tool.bundle.YZVRJ4MO.js              15.62 Kb
├─ erpnext-web.bundle.J434WXAN.js                           3.69 Kb
├─ erpnext.bundle.43CCVQ3Q.js                               156.07 Kb
├─ hierarchy-chart.bundle.BDOAMLZ5.js                       197.69 Kb
├─ item-dashboard.bundle.NUWQRT3L.js                        10.37 Kb
├─ point-of-sale.bundle.X5KQUEAM.js                         89.69 Kb
└─ marketplace.bundle.ZZXYQ5G7.js                           275.98 Kb

erpnext/dist/css/
├─ erpnext-web.bundle.W6F2A5VG.css                          9.36 Kb
├─ erpnext.bundle.PCCETAUU.css                              44.89 Kb
└─ erpnext_email.bundle.GFE2VM7P.css                        0.50 Kb

erpnext/dist/css-rtl/
├─ erpnext-web.bundle.BJ7B4EMK.css                          9.36 Kb
├─ erpnext.bundle.ZZZH7MJZ.css                              44.88 Kb
└─ erpnext_email.bundle.PNG2YP3K.css                        0.50 Kb

 DONE  Total Build Time: 3.064s

Done in 3.59s.

```

## Syntax Error as required.
![image](https://user-images.githubusercontent.com/62703211/136934436-e63ac5e4-f847-4c33-bdd7-804ae42a3711.png)

## avoid using development versions.? 

I would've definitely avoided it, had the proper documentation for the complete installation procedure been provided on the official website instead of searching google/random websites over it. Not even it is mentioned anywhere in any of the installation procedures to choose the version.

**Comment by Fauzdar1** [comment]
Deployed ERPNext on a new server, still same issue, stuck at logo after first login. Do you guys have any solution at all?

**Comment by Fauzdar1** [comment]
Hey @ankush any solutions or should we like many users stuck on the same issue, finally move on from ERPNext?

**Comment by ankush** [comment]
I just setup erpnext on new laptop today and it worked fine. It's entirely possible that you just downloaded it while it was in a broken state (happens sometimes on development version). Updating it might work but best option I can suggest is switching to version-13 branch (stable).

```
bench switch-to-branch version-13
bench --site <yoursitename> reinstall
bench build
```

Note: This will wipe the database.

**Comment by ankush** [comment]
FYI, for technical queries, https://discuss.erpnext.com/ is the appropriate platform.

**Comment by Fauzdar1** [comment]
Hey @ankush Although, the command you provided didn't work because as I've noticed, such commands don't work after installing bench, but, no problem, I've found a solution, the problem was with bench and not with ERPNext. I reinstalled the server, reinstalled bench with specific version and I've faced no issues since then. I've just noticed you had indeed provided me what was needed but I figured it out from a 3rd party website. Thanks.
